client/src/components/primitiveComponents/AlertBoxes.tsx

```diff
@@ -3,30 +3,87 @@ import { AlertDialog, Button, Flex, TextField } from '@radix-ui/themes';
 import { useState } from 'react';
 import { TwilightSelect } from './Primitives';
 
-interface TLAddTimelineAlertProps {
-    year: number;
-    onSubmit: (year: number, timelineEvent: string) => void;
+interface TwilightEditTextAlertProps {
+    labelText: string;
+    textInput: string;
+    objectKey?: string;
+    onSubmit: (newValue: string, objectKey?: string) => void;
+    onDelete?: () => void;
+}
+
+const TwilightEditTextAlert = ({ labelText, textInput, objectKey, onSubmit, onDelete }: TwilightEditTextAlertProps) => {
+    const [displayedText, setDisplayedText] = useState(textInput);
+
+    const handleSubmitEntry = () => {
+        if (objectKey !== undefined) {
+            onSubmit(displayedText, objectKey);
+        } else {
+            onSubmit(displayedText);
+        }
+    };
+    return (
+        <AlertDialog.Root>
+            <AlertDialog.Trigger>
+                <Button variant="ghost">{textInput}</Button>
+            </AlertDialog.Trigger>
+            <AlertDialog.Content style={{ maxWidth: 450 }}>
+                <AlertDialog.Title>{labelText}</AlertDialog.Title>
+                <Flex direction="row" justify="center" align="center" gap="3">
+                    <TextField.Root>
+                        <TextField.Input value={displayedText} onChange={(e) => setDisplayedText(e.target.value)} />
+                    </TextField.Root>
+                </Flex>
+
+                <Flex gap="3" mt="4" justify="end">
+                    <AlertDialog.Cancel>
+                        <Button variant="soft" color="gray">
+                            Cancel
+                        </Button>
+                    </AlertDialog.Cancel>
+                    <AlertDialog.Action>
+                        {textInput !== '' && onDelete !== undefined && (
```

Comment:
Not necessary to change, but remember that empty strings and undefined vars are 'falsey', so this could be written as `textInput && onDelete && ...`